pkg/controller/storagecluster/storagecluster.go

```diff
@@ -1259,7 +1259,18 @@ func getDefaultMaxStorageNodesPerZone(
 		return 0, err
 	}
 	numZones := len(zoneMap)
-	storageNodes = uint64(len(nodeList.Items) / numZones)
+	totalPxNodes := 0
+	for _, node := range nodeList.Items {
+		shouldRun, _, err := k8s.CheckPredicatesForStoragePod(&node, cluster, nil)
```

Comment:
Can we add a unit test for case where the node affinity does not match all the nodes and some nodes with taints for which storagecluster does not have tolerations?

Reply:
shall we call nodeShouldRunStoragePod instead. in addition it looks at disable-storage annotation on storagecluster and if a node is being deleted or not.